mesonbuild/backend/ninjabackend.py

```diff
@@ -2647,10 +2647,10 @@ def guess_external_link_dependencies(self, linker, target, commands, internal):
 
         guessed_dependencies = []
         # TODO The get_library_naming requirement currently excludes link targets that use d or fortran as their main linker
-        if hasattr(linker, 'get_library_naming'):
-            search_dirs = tuple(search_dirs) + tuple(linker.get_library_dirs(self.environment))
```

Comment:
I am not sure I understand this. Does mypy raise errors about `hasattr` usage?

Reply:
mypy doesn't (and according to Guido Rossum) probably never will support hasattr, because it doesn't know what the annotations for the function would be.